libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -1332,6 +1332,10 @@ void ContractCompiler::appendModifierOrFunctionCode()
 
 	if (codeBlock)
 	{
+		bool previousChecked = m_context.checkedArithmetics();
+		// TODO test that checks are also applied for initializing state variables
+		m_context.setCheckedArithmetics(true);
```

Comment:
Why is it not read from the block AST node?

Reply:
My idea was that the statement compiler sets the flag when entering and exiting the unchecked block. The reason this code is here is because we are silently exiting the syntactic area of a modifier.

Reply:
So this is less about the block and more about the `_;`

Reply:
Ah ok